homeassistant/helpers/system_info.py

```diff
@@ -1,25 +1,44 @@
 """Helper to gather system info."""
 from __future__ import annotations
 
+from functools import cache
 from getpass import getuser
+import logging
 import os
 import platform
 from typing import Any
 
 from homeassistant.const import __version__ as current_version
 from homeassistant.core import HomeAssistant
 from homeassistant.loader import bind_hass
-from homeassistant.util.package import is_virtual_env
+from homeassistant.util.package import is_docker_env, is_virtual_env
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@cache
+def is_official_image() -> bool:
+    """Return True if Home Assistant is running in an official container."""
+    return os.path.isfile("/OFFICIAL_IMAGE")
+
+
+# Cache the result of getuser() because it can call getpwuid() which
+# can do blocking I/O to look up the username in /etc/passwd.
+cached_get_user = cache(getuser)
 
 
 @bind_hass
 async def async_get_system_info(hass: HomeAssistant) -> dict[str, Any]:
     """Return info about the system."""
+    is_hassio = (
```

Comment:
You're now checking the same thing twice, as `is_hassio` is implemented exactly as the first check https://github.com/home-assistant/core/blob/dev/homeassistant/components/hassio/__init__.py#L384-L389

Reply:
```
 'homeassistant.components.hassio.const': <module 'homeassistant.components.hassio.const' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/const.py'>, 'homeassistant.components.hassio.handler': <module 'homeassistant.components.hassio.handler' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/handler.py'>, 'homeassistant.components.hassio.addon_manager': <module 'homeassistant.components.hassio.addon_manager' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/addon_manager.py'>, 'homeassistant.components.hassio.addon_panel': <module 'homeassistant.components.hassio.addon_panel' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/addon_panel.py'>, 'homeassistant.components.hassio.auth': <module 'homeassistant.components.hassio.auth' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/auth.py'>, 'homeassistant.helpers.discovery_flow': <module 'homeassistant.helpers.discovery_flow' from '/Users/bdraco/home-assistant/homeassistant/helpers/discovery_flow.py'>, 'homeassistant.components.hassio.discovery': <module 'homeassistant.components.hassio.discovery' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/discovery.py'>, 'homeassistant.components.hassio.http': <module 'homeassistant.components.hassio.http' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/http.py'>, 'homeassistant.components.hassio.ingress': <module 'homeassistant.components.hassio.ingress' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/ingress.py'>, 'homeassistant.components.hassio.issues': <module 'homeassistant.components.hassio.issues' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/issues.py'>, 'homeassistant.components.hassio.websocket_api': <module 'homeassistant.components.hassio.websocket_api' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/websocket_api.py'>, 'homeassistant.components.hassio': <module 'homeassistant.components.hassio' from '/Users/bdraco/home-assistant/homeassistant/components/hassio/__init__.py'>
```

Every core/container system has the modules loaded in memory